### DISCUSSION TITLE
Multi examples with description don't display anymore on the 0.99.x release

### DISCUSSION THREAD

**Discussion by YiuRULE** [fastapi/fastapi#9784] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

Taken as-if on the official documentation [here](https://fastapi.tiangolo.com/tutorial/schema-extra-example/#body-with-multiple-examples)

```python

from typing import Annotated

from fastapi import Body, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None


@app.put("/items/{item_id}")
async def update_item(
    *,
    item_id: int,
    item: Annotated[
        Item,
        Body(
            examples=[
                {
                    "name": "Foo",
                    "description": "A very nice Item",
                    "price": 35.4,
                    "tax": 3.2,
                },
                {
                    "name": "Bar",
                    "price": "35.4",
                },
                {
                    "name": "Baz",
                    "price": "thirty five point four",
                },
            ],
        ),
    ],
):
    results = {"item_id": item_id, "item": item}
    return results
```

### Description

When we have an `examples` , we should have a multi-selection of an example, it doesn't seems to be the case anymore.

Current page:

![multiple_examples](https://github.com/tiangolo/fastapi/assets/4019437/ce27ddb3-bdf0-451e-8e9e-7d7ef22786d7)

Page as described in the documentation:

![documentation](https://github.com/tiangolo/fastapi/assets/4019437/1f89aa55-6869-474f-8511-dd8a66afc79e)

With taking a look to the functional tests, we may send a `summary` field with a description, but it seems that it considers it as an unique value instead.

![summary](https://github.com/tiangolo/fastapi/assets/4019437/1f7e0075-b6ab-4397-a5a3-c4e70a12d973)

The documentation should be changed if it was something made on purpose, or potentially we should provide a bugfix.

### Operating System

Linux

### FastAPI Version

0.99.1

### Python Version

Python 3.10.6

**Comment by Kludex** [comment]
Yeah. The example is wrong. PR welcome to fix it. 👍 

(it should be a dictionary)

**Comment by tiangolo** [comment, marked as answer]
Yep, so, here's the thing, in short, your code is fine with the new version and with JSON Schema in the latest version (used by the latest OpenAPI).

This should be a feature request for Swagger UI.

---

More details:

First, a bunch of details in the new docs: https://fastapi.tiangolo.com/tutorial/schema-extra-example/?h=examples#technical-details

The thing is, there are two `examples`, one is at the OpenAPI level (outside), the other is at the JSON Schema level (inside). The new version of FastAPI, to keep in line with Pydantic and the new JSON Schema (that now supports `examples`) now lets the examples live in JSON Schema, the same that would happen if you were using Pydantic alone. And in JSON Schema (and the new version) it's a list of plain examples.

In OpenAPI (outside of JSON Schema), there's an `examples` field, that takes a dictionary, and up until the previous version, FastAPI would extract those examples declared in parameters and would put it in OpenAPI. But that behavior is inconsistent to what would happen if you put `examples` in Pydantic fields, because those examples would end up in JSON Schema directly, not in OpenAPI. And the shape is different, in OpenAPI it's a dict with other metadata, in JSON Schema it's a plain list of pure examples.

So, now, your OpenAPI actually includes the examples correctly with the new format (without extra metadata) but Swagger UI still doesn't support showing all of the examples, it only shows the first one. The feature request there would be show multiple examples from JSON Schema examples and not just the first one, the same way it can show multiple examples when they are added at the OpenAPI level.

---

**2023-08-26 edit**: There's a new parameter `openapi_examples` just for this, to allow declaring examples that show up in Swagger UI while it implements support for multiple JSON Schema examples. It's available now in FastAPI 0.103.0, the new docs are here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/#openapi-specific-examples 🎉

**Reply by YiuRULE** [reply]
Thank you very much for your answer !

> So, now, your OpenAPI actually includes the examples correctly with the new format (without extra metadata) but Swagger UI still doesn't support showing all of the examples, it only shows the first one.

I think adding `examples` as we have now on the latest release was the good think to do and setting the ability to add multiples examples on a field level definitely make sense, but I suppose that if a similar feature would be implemented in the Swagger UI, it's likely that it would be a list of example who correspond to the product of all possible combinations. I think it's a different use-case compared to what the OpenAPI spec try to respond where people can set payload who may make sense for a direct API user.

I understand the technical challenge and to be fully compatible with pydantic is the thing to do, but if this feature plan to be fully dropped, does it makes sense to possibly adding a capability to modify the generated OpenAPI schema of an object ? (kinda lika the `schema_extra` staticmethod, but without any validation where the developer can modify it, on his own risk)

**Reply by tiangolo** [reply]
Yep, I could consider adding another `openapi_examples` parameter that would go to OpenAPI instead of JSON Schema. But I would rather do it after finishing the Pydantic v2 migration. 😅 

Although I think the first action should be requesting them to show more than one example, they are already showing one, so it's probably a sensible request to want to show multiple examples.

**Reply by tiangolo** [reply]
There's a new parameter `openapi_examples` just for this, to allow declaring examples that show up in Swagger UI while it implements support for multiple JSON Schema examples. It's available now in FastAPI 0.103.0, the new docs are here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/#openapi-specific-examples 🎉

**Reply by cfxegbert** [reply]
Is there a way to include multiple examples for the response?

**Reply by AnddyAgudelo** [reply]
@cfxegbert here: https://fastapi.tiangolo.com/az/advanced/additional-responses/

**Reply by cfxegbert** [reply]
> @cfxegbert here: https://fastapi.tiangolo.com/az/advanced/additional-responses/

That does not cover multiple examples of the same response.
